### ISSUE TITLE
Content within stylelint-disable blocks is being updated by fix function

### ISSUE THREAD

**Issue by sprynm** [stylelint/stylelint#3867]
StyleLint in `fix` mode is changing contents within disable comments where it should skip over the block.

> Which rule, if any, is the bug related to?

`/* stylelint-disable */` or `/* stylelint-disable [rule] */` comments

> What CSS is needed to reproduce the bug?

A specific example is:
```scss
/* stylelint-disable value-keyword-case */
$font-family-monospace:       SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace !default;
/* stylelint-enable value-keyword-case */
```
**Run onscreen, the block produces no warnings**. 

With the Auto fix flag, the line becomes:

```scss
/* stylelint-disable value-keyword-case */
$font-family-monospace:       sfmono-regular, menlo, monaco, consolas, "Liberation Mono", "Courier New", monospace !default;
/* stylelint-enable value-keyword-case */
```

Same outcome for just `/* stylelint-disable */` blocks

> What stylelint configuration is needed to reproduce the bug?

I am running as a gulp task:

```javascript
const scssFiles = [path.join(paths.source.SCSS, '*.scss')];

gulp.task('LintFix', function lintFixCssTask() {
        return gulp
            .src(scssFiles, {
                since: gulp.lastRun('LintFix'),
                base: "./"
            })
            .pipe(gulpStylelint({
                failAfterError: false,
                reportOutputDir: path.join(paths.source.SCSS, 'docs'),
                reporters: [{
                        formatter: 'string',
                        console: true,
                        save: 'stylelint-fix.txt'
                    }
                ],
                fix: true
            }))
            .pipe(gulp.dest('./'));
    });
```

The default task is: 
```javascript
    gulp.task('Lint', function lintCssTask() {
        return gulp
            .src(scssFiles, {
                since: gulp.lastRun('Lint')
            })
            .pipe(gulpStylelint({
                failAfterError: false,
                reportOutputDir: path.join(paths.source.SCSS, 'docs'),
                reporters: [{
                        formatter: 'string',
                        save: 'stylelint.txt'
                    }
                ]
            }));
    });
```

> Which version of stylelint are you using?

- gulp -v4
- gulp-stylelint - 8.0.0
- stylelint - 9.9.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

See above: `gulp Lint` and `gulp LintFix`.  These are later run as part of a watch task.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

My files are written in SCSS - I have not tested in plain CSS.

> What did you expect to happen?

Contents within the disable comments should not be changed by the `fix` function specific to what is disabled.

> What actually happened (e.g. what warnings or errors did you get)?

Contents are updated by any and all "fixable" rules.

There are no notices specific or applicable to this issue.

**Comment by jeddy3** [comment]
@sprynm Thank you for the report and for using the template.

This is a known issue and being tracked in https://github.com/stylelint/stylelint/issues/2643. That issue is labelled as "help wanted" and is looking for someone to champion a rewrite of the disable comments feature so that it works with autofix. The disable comment feature was added before the autofix feature and it works in a way that [isn't compatible with autofix](https://github.com/stylelint/stylelint/issues/2643#issuecomment-309304995).

I'll close this issue as a dupe. Feel free to help out with the other one.
